aspects/aspects.go

```diff
@@ -530,6 +547,75 @@ func schemaTypesStr(types []SchemaType) string {
 	return sb.String()
 }
 
+// getValuesThroughPaths takes a match's storage path and unmatched suffix and
+// strips the outer layers of the value to be set so it can be used at the storage
+// path. Parts of the suffix that are placeholders will be expanded based on what
+// keys exist in the value at that point and the mapping will be used to complete
+// the storage path.
+var getValuesThroughPaths = getValuesThroughPathsImpl
```

Comment:
why the var indirection here?

Reply:
Because the function is recursive so assigning an anonymous function to a var would create a loop when parsing the function (it uses itself but the var isn't defined yet).